guidance/models/transformers/_transformers.py

```diff
@@ -174,6 +76,203 @@ def __init__(
+        #     raise
+        except:
+            # Fall back for other exceptions
+            # TODO: (maybe warn?)
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=True, **kwargs
+            )  # fall back to the fast tokenizer
+
+        return tokenizer
+
+    def _byte_tokens(
+        self,
+        transformers_tokenizer: Union[
+            "transformers_package.PreTrainedTokenizer",
+            "transformers_package.PreTrainedTokenizerFast",
+        ],
+    ) -> list[bytes]:
+
+        if (
+            hasattr(transformers_tokenizer, "byte_decoder")
+            and self._byte_decoder_has_all_bytes(
+                transformers_tokenizer.byte_decoder,
+                transformers_tokenizer.get_vocab()
+            )
+        ):
+            return self._byte_tokens_from_byte_decoder(transformers_tokenizer)
+
+        if hasattr(transformers_tokenizer, "sp_model"):
+            return self._byte_tokens_from_sp_model(transformers_tokenizer)
+
+        try:
+            return self._byte_tokens_from_vocab(transformers_tokenizer)
+        except ValueError:
+            pass
+
+        return self._byte_tokens_fallback(transformers_tokenizer)
+
+    def _byte_tokens_from_byte_decoder(
+        self,
+        transformers_tokenizer: Union[
+            "transformers_package.PreTrainedTokenizer",
+            "transformers_package.PreTrainedTokenizerFast",
+        ],
+    ) -> list[bytes]:
+        byte_tokens = [b""] * len(transformers_tokenizer)
+        byte_decoder: dict[str, int] = transformers_tokenizer.byte_decoder
+        for i in range(len(transformers_tokenizer)):
+            byte_coded = bytes(
+                [byte_decoder[c] for c in transformers_tokenizer.convert_ids_to_tokens(i)]
+            )
+            byte_tokens[i] = byte_coded
+        return byte_tokens
+
+    def _byte_tokens_from_sp_model(
+        self,
+        transformers_tokenizer: Union[
+            "transformers_package.PreTrainedTokenizer",
+            "transformers_package.PreTrainedTokenizerFast",
+        ],
+    ) -> list[bytes]:
+        byte_tokens = [b""] * len(transformers_tokenizer)
+        special_tokens_map = {
+            id: token for token, id in transformers_tokenizer.get_added_vocab().items()
+        }
+        space_prefix = "▁".encode()
+        for i in range(len(transformers_tokenizer)):
+            if i in special_tokens_map:
+                byte_coded = special_tokens_map[i].encode()
+            else:
+                byte_coded = re.sub(
+                    rb"<0x(..)>",
+                    lambda x: bytes.fromhex(x[1].decode()),
+                    transformers_tokenizer.sp_model.id_to_piece(i).encode(),
+                )
+            byte_tokens[i] = byte_coded.replace(space_prefix, b" ")
+        return byte_tokens
+
+    def _byte_tokens_from_vocab(
```

Comment:
This path used to call `transformers_tokenizer.get_vocab` and then do nothing with it. It was also only happening if the tokenizer actually had that method. As far as I can tell, it always has that method (again, could be a version thing?)

Reply:
I think originally I may have used that method to get the tokenizer key/value pairs before replacing it with just iterating through a range the same length as the tokenizer, I guess I didn't realize that we just weren't using the vocabulary at all anymore, so I think removing it makes sense if the functionality stays the same.

Reply:
Got it, thanks!